### ISSUE TITLE
Add a Web Crawler

### ISSUE THREAD

**Issue by dadoonet** [dadoonet/fscrawler#689]
We can base our code on https://github.com/yasserg/crawler4j

**Comment by rajaseg** [comment]
@dadoonet, Thank you very much for the plugin. we are using it for crawling local files.

It would be great to include a Web crawler. Is there any timeline to include a web crawler?

**Comment by dadoonet** [comment]
I honestly don't know if I will support this as Elastic now has this feature available with the basic license.

**Comment by rajaseg** [comment]
> I honestly don't know if I will support this as Elastic now has this feature available with the basic license.

@dadoonet, Yes Elastic has Web Crawle for Site Search. It would be good to have it for Workplace Search. Also, I don't think we can crawl authenticated sites using it.

**Comment by rajaseg** [comment]
Please let me know your availability to discuss my issue.

**Comment by dadoonet** [comment]
What is your issue?

**Comment by rajaseg** [comment]
I am trying to crawl the site, https://web.microsoftstream.com/ using Web Crawler. I added the domain and authentication details.

I got below error while adding the above domain

![image](https://user-images.githubusercontent.com/633479/214602470-6baacf94-bed6-4e3b-b038-c5197f03f7e6.png)
I selected "Ignore validation failures and continue"

No Documents found when I crawl the domain.

**Comment by dadoonet** [comment]
Could you ask in discuss.elastic.co?

**Comment by dadoonet** [comment]
#1820 will probably remove workplace search so we don't need to fix this one specifically for workplace search. And elasticsearch has now a webcrawler so probably this is not needed.

@rajasekhar-gundala WDYT?
